### PR TITLE
Add a new keyboard called Rosa

### PR DESCRIPTION
**Pull request by bleeCS**
## Description

Adding a new keyboard.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

*  N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/illusion/rosa/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Rosa
+
+![Rosa](https://i.imgur.com/ATFjY5f.jpg)
```
Comment (fauxpark) [inline keyboards/illusion/rosa/README.md:3]:
```suggestion
![Rosa](https://i.imgur.com/ATFjY5fh.jpg)
```

File: keyboards/illusion/rosa/rosa.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    setPinOutput(B3); // caps lock indicator LED pin
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if (res) {
+        writePin(B3, !led_state.caps_lock);
+    }
+    return res;
+}
```
Comment (fauxpark) [inline keyboards/illusion/rosa/rosa.c:30]: Can be replaced with a simple define in config.h: https://docs.qmk.fm/#/feature_led_indicators

File: keyboards/illusion/rosa/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/illusion/rosa/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/illusion/rosa/rosa.c (hunk 2 above)
Comment (bleeCS) [reply keyboards/illusion/rosa/rosa.c]: I've never done LED indicators in config.h like this, but I've made what I think are the appropriate changes: 02cd17b2b8a9203885e12aceb260237cce27b287

**Review by fauxpark — APPROVED** [review]

File: keyboards/illusion/rosa/keymaps/oggi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/illusion/rosa/keymaps/oggi/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/illusion/rosa/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Rosa
```
Comment (drashna) [inline keyboards/illusion/rosa/README.md:1]: Could this file be renamed to be all lower case?

**Review by drashna — APPROVED** [review]

**Comment by bleeCS** [comment]
Looking to get a workflow approval, and possibly someone to do a once over of keyboards/illusion/rosa/info.json .

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**